astroid/_ast.py

```diff
@@ -36,6 +37,7 @@ class ParserModule(NamedTuple):
     context_classes: dict[type[ast.expr_context], Context]
 
     def parse(self, string: str, type_comments: bool = True) -> ast.Module:
+        parse_func: Callable[[str], ast.Module]
```

Comment:
This isn't really correct with the `partial`, but let's use it for now. We're removing/revisiting most of this code anyway when we drop 3.7.

Reply:
What should it be instead? Isn't it a function that takes a string and gives back a module?

Reply:
Well `partial` is its one type I believe, but it should do for now!